WordPress/Jetpack/Classes/System/JetpackWindowManager.swift

```diff
@@ -73,12 +82,43 @@ class JetpackWindowManager: WindowManager {
         self.show(container.makeInitialViewController())
     }
 
-    private func switchToAppUI(for blog: Blog?) {
+    func switchToAppUI(for blog: Blog?) {
         cancellable = nil
         showAppUI(for: blog)
     }
 
-    private var shouldShowMigrationUI: Bool {
-        return FeatureFlag.contentMigration.enabled && AccountHelper.isLoggedIn
+    func handleMigrationFailure(_ error: DataMigrationError) {
+        guard
+            case .dataNotReadyToImport = error,
+            isCompatibleWordPressAppPresent,
+            let schemeUrl = URL(string: AppScheme.wordpressMigrationV1.rawValue)
+        else {
+            showSignInUI()
+            return
+        }
+
```

Comment:
I'll break this out into a function to clean things up.

Reply:
Sounds good!